src/Compilers/Core/Portable/Symbols/IMethodSymbol.cs

```diff
@@ -117,6 +117,19 @@ public interface IMethodSymbol : ISymbol
         /// </summary>
         IMethodSymbol ConstructedFrom { get; }
 
+        /// <summary>
+        /// Indicates whether the method is declared readonly, i.e.
+        /// whether 'this' is readonly in the scope of the method.
+        /// See also <see cref="IsEffectivelyReadOnly"/>
+        /// </summary>
+        bool IsDeclaredReadOnly { get; }
```

Comment:
Do properties need this API as well?

Reply:
I should have noted: https://github.com/dotnet/roslyn/issues/34213#issuecomment-476897225
I think we may ship the feature without adding any new API to `IPropertySymbol`.